### PR TITLE
updated README

### PR DESCRIPTION
**Pull request by lwwmanning**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,28 +5,45 @@
 [![Documentation](https://docs.rs/vortex-array/badge.svg)](https://docs.rs/vortex-array)
 [![PyPI - Python Version](https://img.shields.io/pypi/pyversions/vortex-array)](https://pypi.org/project/vortex-array/)
 
-Vortex is an Apache Arrow-compatible toolkit for working with compressed array data. We are using Vortex to develop a
-next-generation columnar file format for multidimensional arrays.
+Vortex is a toolkit for working with compressed Apache Arrow arrays in-memory, on-disk, and over-the-wire.
+
+Vortex is designed to be to columnar file formats what Apache DataFusion is to query engines (or, analogously,
```
Comment (danking) [inline README.md:10]: Concise and roughly same meaning without “designed to be”

File: README.md [hunk 2]
```diff
@@ -5,28 +5,45 @@
...
+what LLVM + Clang are to compilers): a highly extensible & extremely performant *framework* for building a modern
```
Comment (danking) [inline README.md]: I prefer fast to performant personally.

File: README.md [hunk 3]
```diff
@@ -5,28 +5,45 @@
...
+columnar file format, with a state-of-the-art, "batteries included" reference implementation that we're tentatively
+calling "Vinyl".[1]
+
+Vinyl is an aspiring successor to Apache Parquet, with dramatically faster random access reads (100-200x faster)
+and scans (2-10x faster), while preserving approximately the same compression ratio and write throughput. It will also support very wide
+tables (at least 10s of thousands of columns) and (in the future) on-device decompression on GPUs.
```
Comment (danking) [inline README.md]: I’m hoping for millions; is that too speculative to include here?

File: README.md [hunk 4]
```diff
@@ -5,28 +5,45 @@
...
+The major components of Vortex are:
 
 * **Logical Types** - a schema definition that makes no assertions about physical layout.
-* **Encodings** - a pluggable set of physical layouts. Vortex ships with several state-of-the-art lightweight
-  compression codecs that have the potential to support GPU decompression.
-* **Compression** - recursive compression based on stratified samples of the input.
-* **Compute** - basic compute kernels that can operate over compressed data. Note that Vortex does not intend to become
-  a full-fledged compute engine, but rather to provide the ability to implement basic compute operations as may be
-  required for efficient scanning & pushdown operations.
+* **Zero-Copy to Arrow** - "canonicalized" (i.e., fully decompressed) Vortex arrays can be zero-copy converted to/from Apache Arrow arrays.
```
Comment (danking) [inline README.md:27]: These read more like features than components now. Maybe change the lead-in sentence?

File: README.md [hunk 5]
```diff
@@ -5,28 +5,45 @@
...
 * **Statistics** - each array carries around lazily computed summary statistics, optionally populated at read-time.
   These are available to compute kernels as well as to the compressor.
-* **Serde** - zero-copy serialization. Useful as a building block in creating IPC or file formats that contain
-  compressed arrays.
+* **Serialization** - Zero-copy serialization, both for IPC and for file formats.
+* **Vinyl Columnar File Format (in progress)** - A modern file format that uses the Vortex serde library to store compressed array data.
+  Optimized for random access reads and extremely fast scans; an aspiring successor to Apache Parquet.
+
+
+[1]: Because like its predecessor, it's a type of [flooring](https://en.wikipedia.org/wiki/Parquet), but it is much better for
```
Comment (danking) [inline README.md]: ❤️

File: README.md [hunk 6]
```diff
@@ -5,28 +5,45 @@
...
+* **Cascading Compression** - data can be recursively compressed with multiple nested encodings.
+* **Pluggable Compression Strategies** - the built-in Compressor is based on BtrBlocks, but other strategies can trivially be used instead.
+* **Compute** - basic compute kernels that can operate over encoded data. Note that Vortex does not intend to become
+  a full-fledged compute engine, but rather to implement lower-level compute operations required for efficient scanning & pushdown.
 * **Statistics** - each array carries around lazily computed summary statistics, optionally populated at read-time.
   These are available to compute kernels as well as to the compressor.
-* **Serde** - zero-copy serialization. Useful as a building block in creating IPC or file formats that contain
-  compressed arrays.
+* **Serialization** - Zero-copy serialization, both for IPC and for file formats.
+* **Vinyl Columnar File Format (in progress)** - A modern file format that uses the Vortex serde library to store compressed array data.
```
Comment (danking) [inline README.md]: Do we have plans for a native partitioned file format? Maybe that’s what Layouts can achieve? That’s something I’d be looking for if I came to the repo a year ago wondering how to replace Hail’s custom format for a dataset with 100k partitions.

**Review by danking — COMMENTED** [review]
I think my general bias is against superlatives, but I didn’t comment on each one.

File: README.md (hunk 6 above)
Comment (robert3005) [reply README.md]: We will extend vinyl to be multi file but that will be column wise and not row wise. If you have good pruning and random access why have physical file partitioning?

File: README.md (hunk 3 above)
Comment (robert3005) [reply README.md]: We already lazy deserialize schema so might be worth trying to benchmark the overhead of arbirtrary number of columns vs read of selected few. I think layouts get bigger and we have to read them all.

File: README.md (hunk 2 above)
Comment (robert3005) [inline README.md]: Clang is the C++ compiler using LLVM. In this analogy it's just the LLVM

**Review by robert3005 — APPROVED** [review]

**Pull request merged**